src/plugins/intel_npu/src/plugin/npuw/llm_compiled_model.cpp

```diff
@@ -374,11 +381,10 @@ bool remove_empty_kv_inputs(std::shared_ptr<ov::Model> model) {
 }
 }  // namespace
 
-// testability - should we have interface for that or move matchers to another file?
-bool optimize_value_tensors(std::shared_ptr<ov::Model> model);
-bool optimize_value_tensors(std::shared_ptr<ov::Model> model) {
+namespace ov::npuw::util {
+bool optimize_value_tensors(std::shared_ptr<ov::Model> model, bool isPrefill) {
     ov::pass::GraphRewrite rewr;
-    rewr.add_matcher<ScaledDotProductAttentionDecomposition>();
+    rewr.add_matcher<ScaledDotProductAttentionDecomposition>(isPrefill);
```

Comment:
why it must be prefill only? Per our old data, I believe the kvcache model also benefited from switching to F16 activations and there may be the same issue with Add(32) performance due to an extra convert

Reply:
no, generate model degrade if we switch to FP32 precision - due-to smaller tensors, the tiling there i guess is more "suitable", that is why we intentionally changed that, - worth rechecking though one more time on all models

Reply:
generate model is using F32 tensors already in between subgraphs. Actually in our past data models improved their token rates up to the `BEST_PERF` level after F16IC was enabled for those